nrf52: supermini_nrf52840 fails to build — FastPin redefinition between PCA10056 default and TARGET_SUPERMINI_NRF52840

## Symptom

`bash compile supermini_nrf52840` fails on master with:

```
.../platforms/arm/nrf52/fastpin_arm_nrf52_variants.h:1012:10: error: #error "Cannot define more than one board at a time"
.../platforms/arm/nrf52/fastpin_arm_nrf52_variants.h:34:26: error: redefinition of 'class fl::FastPin<0>'
.../platforms/arm/nrf52/fastpin_arm_nrf52_variants.h:34:26: error: redefinition of 'class fl::FastPin<1>'
... (one per pin)
```

CI workflows affected:
- [`nrf52840_supermini`](https://github.com/FastLED/FastLED/actions/workflows/build_nrf52840_supermini.yml)

## Root cause

`SUPERMINI_NRF52840` in `ci/boards.py` reuses `real_board_name="nrf52840_dk_adafruit"` to inherit a working BSP. The Adafruit BSP for that board auto-defines `ARDUINO_NRF52840_PCA10056`, so the PCA10056 pin-map block in `fastpin_arm_nrf52_variants.h` runs first and defines `fl::FastPin<0..>`. Then the `TARGET_SUPERMINI_NRF52840` block (added in #2422 / PR #2422) runs second and redefines the same templates — triggering both the `#error "Cannot define more than one board"` guard and the C++ redefinition diagnostics.

`TARGET_NICE_NANO_V2` and `TARGET_NRFMICRO` ride the same `nrf52840_dk_adafruit` BSP, so the bug would also fire on those community variants the moment they're exercised.

## Fix

Gate the `ARDUINO_NRF52840_PCA10056` block to defer when any of those community `TARGET_*` overrides is in scope, so the explicit user-provided override wins over the BSP-supplied default. Single-file change to `src/platforms/arm/nrf52/fastpin_arm_nrf52_variants.h`.

--- a/.github/workflows/build_nrf52840_supermini.yml
+++ b/.github/workflows/build_nrf52840_supermini.yml
@@ -5,6 +5,7 @@ concurrency:
   cancel-in-progress: ${{ github.event_name == 'pull_request' }}
 
 on:
+  workflow_dispatch:
   push:
     branches:
       - master

--- a/src/platforms/arm/nrf52/fastpin_arm_nrf52_variants.h
+++ b/src/platforms/arm/nrf52/fastpin_arm_nrf52_variants.h
@@ -7,6 +7,20 @@
 // use this to determine if found variant or not (avoid multiple boards at once)
 #undef __FASTPIN_ARM_NRF52_VARIANT_FOUND
 
+// Sentinel: set when the user (or ci/boards.py) has opted in to a community
+// board via TARGET_*. These boards reuse a stock Adafruit BSP (typically
+// nrf52840_dk_adafruit), which injects its own ARDUINO_NRF52840_* macro. The
+// sentinel lets the BSP-default pin-map blocks below defer to the explicit
+// TARGET_* block so the user's override always wins. Add new TARGET_* macros
+// to the OR-chain when introducing new community-board variants. See #2422
+// for the original report and #2626 for the redefinition bug.
+#undef __FASTLED_NRF52_USER_TARGET_OVERRIDE
+#if defined(TARGET_SUPERMINI_NRF52840) || \
+    defined(TARGET_NICE_NANO_V2)      || \
+    defined(TARGET_NRFMICRO)
+    #define __FASTLED_NRF52_USER_TARGET_OVERRIDE
+#endif
+
 
 //
 // BOARD_PIN can be either the pin portion of a port.pin, or the combined NRF_GPIO_PIN_MAP() number.
@@ -346,14 +360,13 @@
 
 // Adafruit Bluefruit on nRF52840DK PCA10056
 // From https://www.adafruit.com/package_adafruit_index.json
-// Community boards (SuperMini / nice!nano v2 / nRFMicro) reuse the
-// nrf52840_dk_adafruit BSP, which auto-defines ARDUINO_NRF52840_PCA10056.
-// When one of those TARGET_* overrides is in scope, defer to its dedicated
-// block below instead of using the PCA10056 pin map (see issue #2422).
-#if defined (ARDUINO_NRF52840_PCA10056) && \
-    !defined(TARGET_SUPERMINI_NRF52840) && \
-    !defined(TARGET_NICE_NANO_V2) && \
-    !defined(TARGET_NRFMICRO)
+// Community boards reuse the nrf52840_dk_adafruit BSP, which auto-defines
+// ARDUINO_NRF52840_PCA10056. When the user opted into one of those community
+// variants via a TARGET_* macro, defer to its dedicated block below instead
+// of running the PCA10056 pin map. Why guard in the header instead of issuing
+// -UARDUINO_NRF52840_PCA10056 from ci/boards.py: a header-side guard also
+// works for non-PIO builds where the user sets TARGET_* by hand.
+#if defined (ARDUINO_NRF52840_PCA10056) && !defined(__FASTLED_NRF52_USER_TARGET_OVERRIDE)
     #if defined(__FASTPIN_ARM_NRF52_VARIANT_FOUND)
         #error "Cannot define more than one board at a time"
     #else

--- a/src/platforms/arm/nrf52/fastpin_arm_nrf52_variants.h
+++ b/src/platforms/arm/nrf52/fastpin_arm_nrf52_variants.h
@@ -346,13 +346,20 @@
 
 // Adafruit Bluefruit on nRF52840DK PCA10056
 // From https://www.adafruit.com/package_adafruit_index.json
-#if defined (ARDUINO_NRF52840_PCA10056)
+// Community boards (SuperMini / nice!nano v2 / nRFMicro) reuse the
+// nrf52840_dk_adafruit BSP, which auto-defines ARDUINO_NRF52840_PCA10056.
+// When one of those TARGET_* overrides is in scope, defer to its dedicated
+// block below instead of using the PCA10056 pin map (see issue #2422).
+#if defined (ARDUINO_NRF52840_PCA10056) && \
+    !defined(TARGET_SUPERMINI_NRF52840) && \
+    !defined(TARGET_NICE_NANO_V2) && \
+    !defined(TARGET_NRFMICRO)
     #if defined(__FASTPIN_ARM_NRF52_VARIANT_FOUND)
         #error "Cannot define more than one board at a time"
     #else
         #define __FASTPIN_ARM_NRF52_VARIANT_FOUND
     #endif
-    
+
     #if defined(USE_ARDUINO_PIN_NUMBERING)
         #error "Define of `USE_ARDUINO_PIN_NUMBERING` has known errors in pin mapping -- select different mapping"
     #elif defined(FASTLED_NRF52_USE_ARDUINO_UNO_R3_HEADER_PIN_NUMBERING)

